crates/evm/traces/src/identifier/signatures.rs

```diff
@@ -95,59 +95,72 @@ impl SignaturesIdentifier {
-        if self.unavailable.contains(identifier) {
-            return None
-        }
-
-        let map = match selector_type {
+    ) -> Vec<Option<T>> {
+        let cache = match selector_type {
             SelectorType::Function => &mut self.cached.functions,
             SelectorType::Event => &mut self.cached.events,
         };
 
-        let hex_identifier = hex::encode_prefixed(identifier);
-
-        if !self.offline && !map.contains_key(&hex_identifier) {
-            if let Ok(signatures) =
-                self.sign_eth_api.decode_selector(&hex_identifier, selector_type).await
-            {
-                if let Some(signature) = signatures.into_iter().next() {
-                    map.insert(hex_identifier.clone(), signature);
+        let hex_identifiers: Vec<String> = identifiers.iter().map(hex::encode_prefixed).collect();
+
+        if !self.offline {
+            let query: Vec<_> = hex_identifiers
+                .iter()
+                .filter(|v| !cache.contains_key(v.as_str()))
+                .filter(|v| !self.unavailable.contains(v.as_str()))
+                .collect();
+
+            if let Ok(res) = self.sign_eth_api.decode_selectors(&query, selector_type).await {
+                for (hex_id, selector_result) in query.into_iter().zip(res.into_iter()) {
+                    let mut found = false;
+                    if let Some(decoded_results) = selector_result {
+                        if let Some(decoded_result) = decoded_results.into_iter().next() {
+                            cache.insert(hex_id.clone(), decoded_result);
+                            found = true;
+                        }
+                    }
+                    if !found {
+                        self.unavailable.insert(hex_id.clone());
+                    }
                 }
             }
         }
 
-        if let Some(signature) = map.get(&hex_identifier) {
-            return get_type(signature).ok()
-        }
-
-        self.unavailable.insert(identifier.to_vec());
-
-        None
+        hex_identifiers
+            .iter()
+            .map(|v| match cache.get(v) {
+                Some(v) => get_type(v).ok(),
+                None => None,
+            })
+            .collect()
     }
 
-    /// Returns `None` if in offline mode
-    fn ensure_not_offline(&self) -> Option<()> {
-        if self.offline {
-            None
-        } else {
-            Some(())
-        }
+    /// Identifies `Function`s from its cache or `https://api.openchain.xyz`
+    pub async fn identify_functions(
+        &mut self,
+        identifiers: &[impl AsRef<[u8]>],
+    ) -> Vec<Option<Function>> {
+        self.identify(SelectorType::Function, identifiers, get_func).await
     }
 
     /// Identifies `Function` from its cache or `https://api.openchain.xyz`
-    pub async fn identify_function(&mut self, identifier: &[u8]) -> Option<Function> {
-        self.ensure_not_offline()?;
```

Comment:
We want to use results from loaded disk cache, network requests will be ignored in `identify()` function